Tools/MiniBrowser/haiku/App.cpp

```diff
@@ -248,8 +257,24 @@ void App::ChangeUrl(BMessage* message)
 	m_url->SetText(str.String());
 	m_url->Looper()->Unlock();
 }
-void 
-App::testLoader()
+
+void App::LoadingProgress(double value)
```

Comment:
This should all go into the MiniBrowser window instead. It will avoid Lock() and Unlock() all over the place and let the BApplication thread clean of a lot of the WebKit event work (which comes from BWebView, and should be a concern of the window it is attached to)

Reply:
So this can be fixed by passing onto correct BLooper right instead of posting message to be_app?

Reply:
Yes, I put a comment at the place where the message is sent as well.